File: app/src/main/java/com/example/pam_app/view/IncomeView.java
package com.example.pam_app.view;

import com.example.pam_app.model.Income;

import java.util.List;

public interface IncomeView {

    void bind(final List<Income> incomeList, final Double incomeLeft);
    void onIncomeAdded(final Income income);
    void onBucketEntryAdded(final Double amount);
    void setUpIncomeLeftText(final boolean isPositive);
}

File: app/src/main/java/com/example/pam_app/view/IncomeView.kt
package com.example.pam_app.view

import com.example.pam_app.model.Income
import kotlin.jvm.Synchronized

interface IncomeView {
    fun bind(incomeList: List<Income>?, incomeLeft: Double)
    fun onIncomeAdded(income: Income?)
    fun onBucketEntryAdded(amount: Double?)
    fun setUpIncomeLeftText(isPositive: Boolean)
}